system/docker/prepare_client.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
```

Comment:
as mentioned, splitting the above prepare script into these two components allow the GHA to build/cache the node and client docker images separately.